### ISSUE TITLE
Insert link: double Enter removes a link because the link text is automatically highlighted after a link is added

### ISSUE THREAD

**Issue by hedgefield** [WordPress/gutenberg#7980]
**Describe the bug**
Pressing Enter twice when adding a link to text makes the linked text disappear and adds a new block.

**To Reproduce**
1. Start a new post in Gutenberg
2. Write something random in a paragraph block
3. Select part of that text and click the Link button in the toolbar
4. Enter some link in the field
5. Press Enter - the link is 'confirmed'
6. Press Enter again - the linked text disappears, and anything that came after it is pushed into a new block.
7. Pressing CTRL-Z then puts back the linked text, but duplicates what came after, and doesn't remove the new block.

**Expected behavior**
I'm not entirely sure, I think I would prefer pressing Enter just once to confirm the link and then close the popover with the edit button. At least to be able to press Escape to close that popover. At the least for the second Enter to close the popover and do nothing else.

**Screenshots**
![jul-16-2018 16-31-41](https://user-images.githubusercontent.com/6073772/42764414-d65bd9a2-8915-11e8-9f07-aa4fb573d3e9.gif)

Gutenberg 3.2
Chrome 67.0.3396.99
MacOS 10.13.5

**Comment by designsimply** [comment]
I'm not really sure either about expected behavior for this one! What is happening is that the link text is automatically highlighted after a link is applied to text and, in that context, it _does_ seem logical to me that pressing enter while text is highlighted will remove the text and insert a line break. It may be a tough case because it's the kind of interaction where there may be a range of differing preferences/opinions. 🙂  Perhaps not auto-selecting link text after a link is added would do the trick? Also, I agree with you that the escape key should dismiss the floating link toolbar!

**Video:** [58s](https://cloudup.com/cSZpSfN4VXF)
<sup>Seen at http://alittletestblog.com/wp-admin/post.php?post=13999&action=edit running WordPress 4.9.7 and Gutenberg 3.2.0 using Firefox 61.0.1 on macOS 10.13.5.</sup>

![link-text-automatically-highlighted](https://user-images.githubusercontent.com/1119271/42832947-13163e06-89b0-11e8-90d4-c1a0f6f9dc5c.png)

**Comment by mcsf** [comment]
@designsimply's assessment is fair, though we should do our best to provide good writing flow. Labeling.

**Comment by designsimply** [comment]
Question on my mind… what do most people do next after adding a link? 🙂

**Comment by designsimply** [comment]
Closed #8257 as a duplicate.

I think it would make sense to place the caret outside the link just after it is inserted because that is where the caret is expected to be if you are adding a link as you are writing and if you are adding a link to already-written content then I think it's a fair expectation that you would need to click on the link again in order to edit it again.

**Comment by karmatosed** [comment]
This seems a valid change, going to remove the design feedback as yes we should get resolved.

**Comment by ellatrix** [comment]
I don't get the problem here. When you select any text in any editor and press enter, you delete the selected text and create a new line. This is expected behaviour.

The same happens when you make something bold or apply any other formatting. The selection doesn't change. I don't see why the link format should get an exception.

**Comment by ellatrix** [comment]
I don't think we should optimise for accidental user key presses.

This is the same for any other format. Pressing `enter` twice:

![bold](https://user-images.githubusercontent.com/4710635/54205452-a5fca980-44d6-11e9-8746-a49a5a6a44ef.gif)

**Comment by designsimply** [comment]
The problem is that the end user expects to be able to enter a link and then continue typing immediately but cannot without deleting the link text. The keypress is intentional in that case. I tested the same steps in Google Docs and when I insert a link and press Enter two times it does not delete the link I just created. I personally like the suggested change because it makes sense to me in a writing flow that I assume is common. However, I'm okay to close if you don't see value in changing the current behavior, although I still want to ask you to consider the typical writing flow as you are writing something out and adding a link. 😊

**Comment by ellatrix** [comment]
> The problem is that the end user expects to be able to enter a link and then continue typing immediately but cannot without deleting the link text.

@designsimply I'm still a bit confused about the writing flow you're referring to. I _do_ consider the typical writing flow.

Here's how it goes in my mind:

1. The user writes something.
2. The user wants to add a link. It's only possible to add a link to something that already been written.
3. The user selects the piece of text that needs to be linked and links it.
4. The user wants to (1) see that the correct thing happened and (2) move on as quickly as possible.

By moving the caret to the end, it's no longer possible to see if the right thing happened (I can't see the link in the link toolbar) and it only enables me to move on quicker _if_ I wish to continue writing _and_ the linked text happens to end where I wish to continue writing. I'm not so sure if it's common to write something, stop at the end of the text you want to link, link it, and then continue where you left off. Moreover, if you do, you're used to having to select the text by either keyboard of mouse, and unselecting it doesn't seem that big of a deal.

> I tested the same steps in Google Docs and when I insert a link and press Enter two times it does not delete the link I just created.

I know Google Docs does it, but it seems to be the odd one this time. Most editors seem to leave the link selected, even the old WordPress editor.

To summarise, I would leave it as is because:

1. You can verify that the link has been added correctly (it appears in the link toolbar).
2. It sounds like it would optimise for a minority of the cases.
3. We'd have to create special behaviour for the link format (I like that all formats have the same behaviour).
4. It seems to be common behaviour (except for Google Docs).
5. It's been like this for ages, the old WordPress editor did the same.

**Comment by paaljoachim** [comment]
Hey Tim @hedgefield 

Is this issue still valid?

**Comment by hedgefield** [comment]
Hey Paal, checked again and it's fixed now, the link remains intact. Closing!
